ui/pages/onboarding-flow/pin-extension/pin-extension.js

```diff
@@ -69,7 +69,9 @@ export default function OnboardingPinExtension() {
     if (selectedIndex === 0) {
       setSelectedIndex(1);
     } else {
-      dispatch(toggleExternalServices(externalServicesOnboardingToggleState));
+      await dispatch(
+        toggleExternalServices(externalServicesOnboardingToggleState),
+      );
```

Comment:
There was a race condition here.

When you disable basic functionality during onboarding, we don't actually flip the setting until this final page.  But the toggle was not awaited, so there was a brief period where:

- completedOnboarding == true
- useExternalServices == true

And then it would quickly flip to false.  But with this change useExternalServices will be false immediately upon leaving onboarding.